src/js/backup_restore.js

```diff
@@ -34,6 +34,9 @@ function configuration_backup(callback) {
         if (semver.gte(CONFIG.apiVersion, "1.15.0")) {
             profileSpecificData.push(MSPCodes.MSP_RC_DEADBAND);
         }
+        if (semver.gte(CONFIG.apiVersion, "1.41.0")) {
```

Comment:
Looks like this has landed in the wrong block - mode ranges are not profile specific.

Reply:
ok, will have a look at that.

Reply:
@mikeller: would this then be true of `MSP_MODE_RANGES` defined in the original `profileSpecificData` on line 26? That's why I added EXTRAS here.

Reply:
I see. The background to this is that in the early days, what is now known as 'PID profiles' was called 'profiles', and included a lot more settings, including mode ranges (probably originally intended to allow a user to use the same flight controller on different models). At this stage, 'profiles' included modes. `MSP_MODE_RANGES` was left in the profile specific section ever since - this causes the same mode range to be written thrice, but it retains backwards compatibility. For `MSP_MODE_RANGES_EXTRA`, there is no need to do the same, as there are no backwards compatibility considerations.

Reply:
ok, good to know.

Reply:
So, is there going to be a problem with the restore function? The same call to `MSP_SET_MODE_RANGES` contains the EXTRA data, too. I think `MODE_RANGES_EXTRA` will have to be read three times to conserve backwards compatibility.

Reply:
Heh, good point, there is no practical downside to having `MODE_RANGES_EXTRA` written / read three times, so let's go for that.

Reply:
Found a way around loading `MODE_RANGES_EXTRA` three times.